compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -7773,9 +7773,22 @@ public void visit(BLangTypeTestExpr typeTestExpr) {
         if (types.isValueType(expr.getBType())) {
             addConversionExprIfRequired(expr, symTable.anyType);
         }
-        typeTestExpr.expr = rewriteExpr(expr);
-        typeTestExpr.typeNode = rewrite(typeTestExpr.typeNode, env);
-        result = typeTestExpr;
+        if (!typeTestExpr.isNotIsExpr) {
+            typeTestExpr.expr = rewriteExpr(expr);
+            typeTestExpr.typeNode = rewrite(typeTestExpr.typeNode, env);
+            result = typeTestExpr;
+            return;
+        }
+        BLangTypeTestExpr bLangTypeTestExpr = ASTBuilderUtil.createTypeTestExpr(typeTestExpr.pos,
+                typeTestExpr.expr, typeTestExpr.typeNode);
+        BLangGroupExpr bLangGroupExpr = (BLangGroupExpr) TreeBuilder.createGroupExpressionNode();
+        bLangGroupExpr.expression = bLangTypeTestExpr;
+        bLangGroupExpr.setBType(typeTestExpr.getBType());
+        BLangUnaryExpr unaryExpr = ASTBuilderUtil.createUnaryExpr(typeTestExpr.pos);
```

Comment:
There's a createUnaryExpr() method that accepts the position, expression, type, operator, and symbol. It'll be more convenient to use that, since you don't have to specify them separately as you have done in lines 7788-7790.